src/plot.js

```diff
@@ -486,73 +530,133 @@ function nolabel(axis) {
     : Object.assign(Object.create(axis), {label: undefined});
 }
 
-// Unlike facetGroups, which returns groups in order of input data, this returns
-// keys in order of the associated scale’s domains.
-function facetKeys({fx, fy}) {
-  return fx && fy ? cross(fx.domain(), fy.domain()) : fx ? fx.domain() : fy.domain();
+// Returns an array of {x?, y?} objects representing the facet domain.
+function Facets(channelsByScale, options) {
+  const {fx, fy} = Scales(channelsByScale, options);
+  let fxDomain = fx?.scale.domain(); // TODO .sort(ascending)?
+  let fyDomain = fy?.scale.domain(); // TODO .sort(ascending)?
```

Comment:
Do we need to sort here or can I remove these comments?

Reply:
removed